homeassistant/components/melnor/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow for melnor."""
+
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import async_discovered_service_info
+from homeassistant.components.bluetooth.models import BluetoothServiceInfoBleak
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, MANUFACTURER_DATA_START, MANUFACTURER_ID
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for melnor."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovered_address: str
+        self._discovered_addresses: list[str] = []
+
+    def _create_entry(self, address: str) -> FlowResult:
+        """Create an entry for a discovered device."""
+
+        return self.async_create_entry(
+            title=address,
+            data={
+                CONF_ADDRESS: address,
+            },
+        )
+
+    async def async_step_bluetooth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle user-confirmation of discovered device."""
+
+        if user_input is not None:
+            return self._create_entry(self._discovered_address)
+
+        return self.async_show_form(
+            step_id="bluetooth_confirm",
+            description_placeholders={"name": self._discovered_address},
+        )
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> FlowResult:
+        """Handle a flow initialized by Bluetooth discovery."""
+
+        address = discovery_info.address
+
+        await self.async_set_unique_id(address)
+        self._abort_if_unique_id_configured(updates={CONF_ADDRESS: address})
+
+        self._discovered_address = address
+
+        self.context["title_placeholders"] = {"name": address}
+        return await self.async_step_bluetooth_confirm()
+
+    async def async_step_pick_device(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the step to pick discovered device."""
+
+        if user_input is not None:
+            return self._create_entry(user_input[CONF_ADDRESS])
```

Comment:
We need to do an `await self.async_set_unique_id(address, raise_on_progress=False)` and `self._abort_if_unique_id_configured()` here before creating the entry in case they started a user flow but didn't finish and then found it via discovery

Reply:
On it 👍🏻

Reply:
done